.github/workflows/image-canary.yaml

```diff
@@ -0,0 +1,60 @@
+name: ImageCanary
+on:
+  schedule:
+  - cron: '0 */1 * * *'
+jobs:
+  ImageCanary:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write # aws-actions/configure-aws-credentials@v4.0.1
+    if: github.repository == 'aws/karpenter-provider-aws'
+    steps:
+    - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@e3dd6a429d7300a6a4c196c26e071d42e0343502 # v4.0.2
+      with:
+        role-to-assume: arn:aws:iam::${{ vars.READONLY_ACCOUNT_ID }}:role/${{ vars.READONLY_ROLE_NAME }}
+        aws-region: ${{ vars.READONLY_REGION }}
+        role-duration-seconds: 900
+    # Authenticate to public ECR to prevent rate limiting
+    - uses: docker/login-action@e92390c5fb421da1463c202d546fed0ec5c39f20 # v3.1.0
+      with:
+        registry: public.ecr.aws
+        logout: true
+      env:
+        AWS_REGION: us-east-1
+    - name: Pull artifacts
+      env:
+        GH_TOKEN: ${{ github.token }}
+      run: |
+        # Translates a git tag to the corresponding ECR tag. Starting with Karpenter v0.35.0 the leading v is dropped in the ECR tag.
```

Comment:
Is it worth pulling this out into a `hack` file or `script`? Makes it easier to parse for an IDE and works better for stand-alone calls

Reply:
Maybe, wasn't sure if this would be something we'd really be doing as a standalone call. Could go either way here.

Reply:
I'd prefer that you pull it out for testing purposes, if there is ever an issue with the script, much easier to call the executable than having to copy over the script from the GHA